### PR TITLE
#479 Add 0.8.0 workflow example

### PR DESCRIPTION
**Pull request by tortmayr**
- Add the sources for the 0.8.0 workflow example
- Improve example structure
- Add a vscode workspace file for each example
- Add support of the Java code in Vscode

Fixes eclipse-glsp/glsp/issues/479

### CODE REVIEW

File: minimal/glsp-server/.vscode/tasks.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      "type": "shell",
+      "group": {
+        "kind": "build",
+        "isDefault": true
+      },
+      "presentation": {
+        "reveal": "always",
+        "panel": "new"
+      },
+      "command": "yarn ",
```
Comment (ndoschek) [inline minimal/glsp-server/.vscode/tasks.json:17]: This is fixed with https://github.com/eclipse-glsp/glsp-examples/pull/89, but for the sake of completeness:
```suggestion
      "command": "mvn clean verify",
```

File: workflow/glsp-server/.vscode/tasks.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      "type": "shell",
+      "group": {
+        "kind": "build",
+        "isDefault": true
+      },
+      "presentation": {
+        "reveal": "always",
+        "panel": "new"
+      },
+      "command": "yarn ",
```
Comment (ndoschek) [inline workflow/glsp-server/.vscode/tasks.json:17]:
```suggestion
      "command": "mvn clean verify",
```

File: minimal/glsp-client/package.json [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  ],
+  "scripts": {
+    "prepare": "lerna run prepare",
+    "watch": "lerna run --parallel watch",
+    "build": "yarn simpleInstall && lerna run build",
+    "lint": "lerna run lint --",
+    "simpleInstall": "yarn install --ignore-scripts",
+    "rebuild:browser": "theia rebuild:browser",
+    "start": "yarn rebuild:browser && yarn --cwd minimal-browser-app start",
+    "start:external": "yarn rebuild:browser && --cwd minimal-browser-app start:external"
```
Comment (ndoschek) [inline minimal/glsp-client/package.json:34]: This is fixed with https://github.com/eclipse-glsp/glsp-examples/pull/89, but for the sake of completeness:
```suggestion
    "start:external": "yarn rebuild:browser && yarn --cwd minimal-browser-app start:external"
```

File: workflow/glsp-client/package.json [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    }
+  ],
+  "scripts": {
+    "prepare": "lerna run prepare",
+    "watch": "lerna run --parallel watch",
+    "build": "yarn install --ignore-scripts && lerna run build",
+    "lint": "lerna run lint --",
+    "rebuild:browser": "theia rebuild:browser",
+    "start": "yarn rebuild:browser && yarn --cwd workflow-browser-app start",
+    "start:external": "yarn rebuild:browser && --cwd workflow-browser-app start:external"
```
Comment (ndoschek) [inline workflow/glsp-client/package.json:33]:
```suggestion
    "start:external": "yarn rebuild:browser && yarn --cwd workflow-browser-app start:external"
```

File: workflow/glsp-client/workflow-browser-app/package.json [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    "@theia/workspace": "1.4.0",
+    "@eclipse-glsp-examples/workflow-theia": "0.8.0"
+  },
+  "devDependencies": {
+    "@theia/cli": "1.4.0"
+  },
+  "scripts": {
+    "prepare": "theia build --mode development",
+    "start": "theia start --WF_GLSP=5007 --root-dir=../workspace",
+    "start:debug": "theia start --WF_GLSP=5007  --root-dir=../workspace --loglevel=debug",
```
Comment (ndoschek) [inline workflow/glsp-client/workflow-browser-app/package.json:26]: This should be named `start:external` to align with the task call (and the minimal example). 
```suggestion
    "start:external": "theia start --WF_GLSP=5007 --root-dir=../workspace --DEBUG",
```

**Review by ndoschek — CHANGES_REQUESTED** [review]
Thanks Tobias looks good to me overall!
There are some minor issues with the VSCode tasks, I added some suggestions inline.

- The workflow example is not consuming the new styling yet, will we update the workflow example to 0.9.0 in a follow-up?
- Regarding the file headers of the changed files, we should update them to `Copyright (c) 20xx-2022 ...`

**Comment by tortmayr** [comment]
> The workflow example is not consuming the new styling yet, will we update the workflow example to 0.9.0 in a follow-up?

It will be updated to 0.9.0 in a follow-up. I'm working on it right now. However, 0.9.0 also does not contain the new styling. The new styling is currently only part of the nightly releases (next). We will have to wait for the 0.10.0 release to consume it here. (This example repository is intended to stay on the latest release and not any nightly versions).

> Regarding the file headers of the changed files, we should update them to Copyright (c) 20xx-2022 ...
I guess so, I wil take care of it

**Review by ndoschek — APPROVED** [review]
Thanks a lot for the quick update Tobias! LGTM :+1:

**Pull request merged**